mitmproxy/proxy/protocol/tls.py

```diff
@@ -336,6 +336,7 @@ def __call__(self):
             # Peek into the connection, read the initial client hello and parse it to obtain SNI and ALPN values.
             try:
                 self._client_hello = TlsClientHello.from_client_conn(self.client_conn)
+                self.client_conn.sni = self._client_hello.sni
```

Comment:
This is using our own TlsClientHello parser, which may not work under some circumstances. Can we use [`.get_servername()`](http://www.pyopenssl.org/en/stable/api/ssl.html?highlight=get_servername#OpenSSL.SSL.Connection.get_servername) [here](https://github.com/chhsiao90/mitmproxy/blob/7274b3a58f8eb36e68bc4d758c5e191b684d16d9/mitmproxy/proxy/protocol/tls.py#L485) instead?